ql/CMakeLists.txt

```diff
@@ -2327,8 +2328,9 @@ target_compile_options(ql_library PRIVATE
     ${OpenMP_CXX_FLAGS})
 
 target_include_directories(ql_library PUBLIC
-    ${CMAKE_BINARY_DIR}
-    ${CMAKE_SOURCE_DIR}
```

Comment:
Answering your question from the opening comment -- we need both of these dirs for the build of the ql library itself since some of the headers are out-of-tree build artifacts generated either by configure_file or the local cmake scripts which generate headers.  Downstream consumers of the fully built ql would only require the single (merged) installation location of the installed (source/binary) headers.  Not sure if that was the gist of your question.  You're overall direction with this issue is sound.  Keep going.

Reply:
Thanks for the clarification. I'll keep both include dirs as is then.